src/modules/partition/jobs/CreatePartitionJob.cpp

```diff
@@ -32,9 +33,134 @@ CreatePartitionJob::CreatePartitionJob( Device* device, Partition* partition )
+    { "c215d751-7bcd-4649-be90-6627490a4c05", "/usr Verity Partition (32-bit ARM)" },
+    { "6e11a4e7-fbca-4ded-b9e9-e1a512bb664e", "/usr Verity Partition (64-bit ARM/AArch64)" },
+    { "6a491e03-3be7-4545-8e38-83320e0ea880", "/usr Verity Partition (Itanium/IA-64)" },
+    { "933ac7e1-2eb4-4f13-b844-0e14e2aef915", "Home Partition" },
+    { "3b8f8425-20e0-4f3b-907f-1a25a76f98e8", "Server Data Partition" },
+    { "4d21b016-b534-45c2-a9fb-5c16e091fd2d", "Variable Data Partition" },
+    { "7ec6f557-3bc5-4aca-b293-16ef5df639d1", "Temporary Data Partition" },
+    { "0657fd6d-a4ab-43c4-84e5-0933c84b4f4f", "Swap" },
+    { "c12a7328-f81f-11d2-ba4b-00a0c93ec93b", "EFI System Partition" },
+    { "bc13c2ff-59e6-4262-a352-b275fd6f7172", "Extended Boot Loader Partition" },
+    { "0fc63daf-8483-4772-8e79-3d69d8477de4", "Other Data Partitions" },
+    { "ebd0a0a2-b9e5-4433-87c0-68b6b72699c7", "Microsoft basic data" },
+};
+
+static QString
+prettyGptType( const QString& type )
+{
+    return gptTypePrettyStrings.value( type.toLower(), type );
+}
+
+static QString
+prettyGptAttributes( quint64 attributes )
+{
+    QStringList list;
+
+    if ( attributes & ( 1UL << 0 ) )
+    {
+        list += QString( "RequiredPartition" );
+    }
+
+    if ( attributes & ( 1UL << 1 ) )
+    {
+        list += QString( "NoBlockIOProtocol" );
+    }
+
+    if ( attributes & ( 1UL << 2 ) )
+    {
+        list += QString( "LegacyBIOSBootable" );
+    }
+
+    for ( qint8 bit = 3; bit < 64; bit++ )
+    {
+        if ( attributes & ( 1UL << bit ) )
+        {
+            list += QString( "Bit" ) + QString::number( bit );
+        }
+    }
+
+    QString string = list.join( QStringLiteral( " and " ) );
+    if ( string.size() > 64 )   
+    {
+        return QString( "0x" ) + QString::number( attributes, 0x10 );
+    }
+
+    return string;
+}
+
+static QString
+prettyGptEntries( const Partition* partition )
+{
+    if ( !partition )
+    {
+        return QString("");
+    }
+
+    QStringList list;
+
+    if ( partition->label().size() )
+    {
+        list += partition->label();
+    }
+
+    if ( partition->uuid().size() )
+    {
+        list += partition->uuid();
+    }
+
+    QString type = prettyGptType( partition->type() );
+    if ( type.size() )
+    {
+        list += type;
+    }
+
+    QString attributes = prettyGptAttributes( partition->attributes() );
+    if ( attributes.size() && attributes != QString( "0" ) )
+    {
+        list += attributes;
+    }
+
+    return list.join( QStringLiteral( ", " ) );
+}
+
 QString
 CreatePartitionJob::prettyName() const
 {
+    if ( CalamaresUtils::Partition::getPartitionTable( m_partition )->type() == PartitionTable::TableType::gpt )
+    {
+        QString entries = prettyGptEntries( m_partition );
+        if ( entries.size() )
+        {
+             entries = QString( " with entries " ) + entries;
+        }
+
+        return tr( "Create new %1MiB partition on %3 (%2)%4." )
```

Comment:
This really needs some translator hints (and maybe a space after `)`?)

Reply:
.. ah, I see that %4 is the bit of the sentence that describes the entries, and if it is not empty, it starts with a space. This is an unpleasant thing to do with translations, so it probably needs some more code to handle the zero / one / many cases of entries.

Reply:
Oh, I was not aware of that.

Reply:
Now I understand why there are such if/else in the code somewhere to deal with cases; that makes the code no pleasant to me lol.

Reply:
Placeholders (or worse, concatenation in hardcoded order – the placeholders can at least be reordered) are generally a possible source of trouble for translations, they should only be used where they make linguistic sense.

I have seen worse abuses in other projects (usually in translator complaints), such as `tr("item") + (n>1 ? "s" : "")`. (Hint: That cannot possibly work, and no, `tr("s")` does not fix it either. Just see what happens if you translate `item` to `entry` in some `en_*`, and that's not even taking into account all the strange things other languages do with plurals.) Not everything that works in English, or even in European languages in general, is safe for translation.

Reply:
We could do this:
```
  // String we already have
  QString overall = tr( "Create new %2MiB partition on %4 (%3) with file system %1." );
  if (!entries.isEmpty()) {
    overall += QStringLiteral(" ") + tr("Special partition entries:") + QStringLiteral(" ") + entries;
  }
  return overall;
```

That makes each call to tr() a single sentence, and avoids translating sentence fragments.